solidity/app/deploy.js

```diff
@@ -0,0 +1,72 @@
+const Eth = require('ethjs')
+const TruffleContract = require('truffle-contract')
+const ABI = require('ethereumjs-abi')
+
+const clUtils = require('./cl_utils.js')
+const clWallet = require('./cl_wallet.js')
+const compile = require('./compile.js')
+
+const TruffleDefaults = {
+  gas: 6721975,
+  gasPrice: 100000000000
+}
+
+function getBytecode (contract) {
+  return contract.evm.bytecode.object.toString()
+}
+
+function contractify (abi, address) {
+  const contract = TruffleContract({
+    abi: abi,
+    address: address
+  })
+  contract.setProvider(clUtils.provider)
+  contract.defaults({
+    from: clWallet.address,
+    gas: TruffleDefaults.gas,
+    gasPrice: TruffleDefaults.gasPrice
+  })
+  return contract.at(address)
+}
+
+function findConstructor (abi) {
+  for (let method of abi) {
+    if (method.type === 'constructor') return method
+  }
+}
+
+function constructorInputTypes (abi) {
+  const types = []
+  for (let input of findConstructor(abi).inputs) {
+    types.push(input.type)
+  }
+  return types
+}
+
+function encodeArgs (unencoded, abi) {
+  if (unencoded.length === 0) {
+    return ''
+  }
+  const buf = ABI.rawEncode(constructorInputTypes(abi), unencoded)
+  return buf.toString('hex')
+}
+
+module.exports = async function deploy (filename) {
+  const unencodedArgs = Array.prototype.slice.call(arguments).slice(1)
```

Comment:
Is there a way to define the variadic `unencodedArgs` in the function definition? It's a little hidden in
```
const unencodedArgs = Array.prototype.slice.call(arguments).slice(1)
```


This is more discoverable

```
module.exports = async function deploy (filename, ...unencodedArgs) {
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters